src/test/java/seedu/address/model/person/CategoryTest.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.model.person;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import seedu.address.testutil.Assert;
+
+public class CategoryTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> new Category(null));
+    }
+
+    @Test
+    public void constructor_invalidSubject_throwsIllegalArgumentException() {
+        String invalidSubject = "";
+        Assert.assertThrows(IllegalArgumentException.class, () -> new Category(invalidSubject));
+    }
+
+    @Test
+    public void isValidCategory() {
+        // null subject
+        Assert.assertThrows(NullPointerException.class, () -> Category.isValidCategory(null));
+
+        // invalid subject
```

Comment:
Remember to change the copy paste comments, I don't think this is checking for `subject` anymore